### ISSUE TITLE
Type of .class expression not supported

### ISSUE THREAD

**Issue by mlangkabel** [ftomassetti/java-symbol-solver#64]
It seems like the type returned by `SomeClass.class` is not supported by now.

**Comment by mlangkabel** [comment]
I think we need another TypeUsage class to support this. Any suggenstions on how to call it? `ClassTypeUsage` may be a little bit confusing.

**Comment by mlangkabel** [comment]
Maybe I shoud give an example to illustrate this case. Consider the following method call:
```
EnumSet.noneOf(SomeClass.class);
```

When solving this call the symbol solver throws this exception:
```
UnsolvedSymbolException{context='ReflectionClassDeclaration{clazz=java.util.EnumSet}', name='Method 'noneOf' with parameterTypes [ReferenceTypeUsage{declaration=JavaParserEnumDeclaration{wrappedNode=public class SomeClass{...}}, typeParameters=[]}]', typeSolver=null}
```

The solver solves the call's argument type as SomeClass, not as Class<SomeClass>

**Comment by mlangkabel** [comment]
Fixed with merge of pull request #65.
